src/main/java/net/robinfriedli/aiode/audio/spotify/SpotifyTrackResultHandler.java

```diff
@@ -69,11 +69,19 @@ public Track getBestResult(String searchTerm, Collection<Track> tracks) {
         }
 
         int bestScore = tracksByScore.keySet().stream().mapToInt(k -> k).max().getAsInt();
-        return tracksByScore
+        Track bestScoringTrack = null;
+
+        tracksByScore
             .get(bestScore)
             .stream()
             .max(Comparator.comparing(Track::getPopularity))
-            .get();
+            .ifPresent(bestTrack -> { bestScoringTrack = bestTrack; });
```

Comment:
Variable used in lambda expression should be final or effectively final 

For reference : https://stackoverflow.com/questions/34865383/variable-used-in-lambda-expression-should-be-final-or-effectively-final

Reply:
Mmmh, so everything that's not an object wrapped object in an object is faulty in Java, interesting. Thanks for the insight!